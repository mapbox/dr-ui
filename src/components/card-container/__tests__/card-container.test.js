import React from 'react';
import renderer from 'react-test-renderer';
import { testCases } from './card-container-test-cases.js';

describe('card-container', () => {
  describe(testCases.basic.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe(testCases.noImage.description, () => {
    let testCase;
    let wrapper;
    let tree;

    beforeEach(() => {
      testCase = testCases.noImage;
      wrapper = renderer.create(
        React.createElement(testCase.component, testCase.props)
      );
      tree = wrapper.toJSON();
    });

    test('renders as expected', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
