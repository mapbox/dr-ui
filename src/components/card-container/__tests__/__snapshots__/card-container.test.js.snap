// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card-container Basic renders as expected 1`] = `
<div>
  <a
    className="unprose mb18 block color-blue-on-hover"
    href="#category"
  >
    <h2
      className="txt-bold unprose"
      id="category"
    >
      Category
       
      <span
        data-swiftype-index="false"
      >
        (
        2
        )
      </span>
    </h2>
  </a>
  <div
    className="grid grid--gut36"
  >
    <div
      className="mb18 col col--12 col--6-ml"
    >
      <a
        className="color-gray-dark transition color-blue-on-hover round clip inline-block w-full unprose pb18"
        href="path"
      >
        <div
          className="relative h120 mb6"
        >
          <div
            className="relative h120 mb12"
            style={
              Object {
                "backgroundImage": "url('/files/simple-map.png')",
                "backgroundPosition": "center",
                "backgroundSize": "100% auto",
              }
            }
          />
        </div>
        <div>
          
          <div
            className="mb6"
          >
            Example one
          </div>
          <div
            className="color-gray color-gray-on-hover"
          >
            description
          </div>
        </div>
      </a>
    </div>
    <div
      className="mb18 col col--12 col--6-ml"
    >
      <a
        className="color-gray-dark transition color-blue-on-hover round clip inline-block w-full unprose pb18"
        href="path"
      >
        <div
          className="relative h120 mb6"
        >
          <div
            className="relative h120 mb12"
            style={
              Object {
                "backgroundImage": "url('/files/simple-map.png')",
                "backgroundPosition": "center",
                "backgroundSize": "100% auto",
              }
            }
          />
        </div>
        <div>
          
          <div
            className="mb6"
          >
            Example two
          </div>
          <div
            className="color-gray color-gray-on-hover"
          >
            description
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`card-container Full width renders as expected 1`] = `
<div>
  <a
    className="unprose mb18 block color-blue-on-hover"
    href="#another-category"
  >
    <h2
      className="txt-bold unprose"
      id="another-category"
    >
      Another category
       
      <span
        data-swiftype-index="false"
      >
        (
        2
        )
      </span>
    </h2>
  </a>
  <div
    className=""
  >
    <div
      className="mb18 border-b border--darken10"
    >
      <a
        className="color-gray-dark transition color-blue-on-hover round clip inline-block w-full unprose pb18"
        href="path"
      >
        
        <div>
          
          <div
            className="mb6"
          >
            Example one
          </div>
          <div
            className="color-gray color-gray-on-hover"
          >
            description
          </div>
        </div>
      </a>
    </div>
    <div
      className="mb18 border-b border--darken10"
    >
      <a
        className="color-gray-dark transition color-blue-on-hover round clip inline-block w-full unprose pb18"
        href="path"
      >
        
        <div>
          
          <div
            className="mb6"
          >
            Example two
          </div>
          <div
            className="color-gray color-gray-on-hover"
          >
            description
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;

exports[`card-container Use cardColSize renders as expected 1`] = `
<div>
  <a
    className="unprose mb18 block color-blue-on-hover"
    href="#another-category"
  >
    <h2
      className="txt-bold unprose"
      id="another-category"
    >
      Another category
       
      <span
        data-swiftype-index="false"
      >
        (
        4
        )
      </span>
    </h2>
  </a>
  <div
    className="grid grid--gut36"
  >
    <div
      className="mb18 col col--12 col--3-ml"
    >
      <a
        className="color-gray-dark transition color-blue-on-hover round clip inline-block w-full unprose pb18"
        href="path"
      >
        <div
          className="relative h120 mb6"
        >
          <div
            className="relative h120 mb12"
            style={
              Object {
                "backgroundImage": "url('/files/simple-map.png')",
                "backgroundPosition": "center",
                "backgroundSize": "100% auto",
              }
            }
          />
        </div>
        <div>
          
          <div
            className="mb6"
          >
            Example one
          </div>
          <div
            className="color-gray color-gray-on-hover"
          >
            description
          </div>
        </div>
      </a>
    </div>
    <div
      className="mb18 col col--12 col--3-ml"
    >
      <a
        className="color-gray-dark transition color-blue-on-hover round clip inline-block w-full unprose pb18"
        href="path"
      >
        <div
          className="relative h120 mb6"
        >
          <div
            className="relative h120 mb12"
            style={
              Object {
                "backgroundImage": "url('/files/simple-map.png')",
                "backgroundPosition": "center",
                "backgroundSize": "100% auto",
              }
            }
          />
        </div>
        <div>
          
          <div
            className="mb6"
          >
            Example two
          </div>
          <div
            className="color-gray color-gray-on-hover"
          >
            description
          </div>
        </div>
      </a>
    </div>
    <div
      className="mb18 col col--12 col--3-ml"
    >
      <a
        className="color-gray-dark transition color-blue-on-hover round clip inline-block w-full unprose pb18"
        href="path"
      >
        <div
          className="relative h120 mb6"
        >
          <div
            className="relative h120 mb12"
            style={
              Object {
                "backgroundImage": "url('/files/simple-map.png')",
                "backgroundPosition": "center",
                "backgroundSize": "100% auto",
              }
            }
          />
        </div>
        <div>
          
          <div
            className="mb6"
          >
            Example one
          </div>
          <div
            className="color-gray color-gray-on-hover"
          >
            description
          </div>
        </div>
      </a>
    </div>
    <div
      className="mb18 col col--12 col--3-ml"
    >
      <a
        className="color-gray-dark transition color-blue-on-hover round clip inline-block w-full unprose pb18"
        href="path"
      >
        <div
          className="relative h120 mb6"
        >
          <div
            className="relative h120 mb12"
            style={
              Object {
                "backgroundImage": "url('/files/simple-map.png')",
                "backgroundPosition": "center",
                "backgroundSize": "100% auto",
              }
            }
          />
        </div>
        <div>
          
          <div
            className="mb6"
          >
            Example two
          </div>
          <div
            className="color-gray color-gray-on-hover"
          >
            description
          </div>
        </div>
      </a>
    </div>
  </div>
</div>
`;
